lib/nmatrix/jruby/nmatrix_java.rb

```diff
@@ -0,0 +1,836 @@
+require 'java'
+require_relative '../../../ext/nmatrix_java/vendor/commons-math3-3.6.1.jar'
+require_relative '../../../ext/nmatrix_java/target/nmatrix.jar'
+
+java_import 'org.apache.commons.math3.linear.ArrayRealVector'
+java_import 'org.apache.commons.math3.linear.RealMatrix'
+java_import 'org.apache.commons.math3.linear.MatrixUtils'
+java_import 'org.apache.commons.math3.linear.DecompositionSolver'
+java_import 'org.apache.commons.math3.linear.LUDecomposition'
+java_import 'org.apache.commons.math3.linear.QRDecomposition'
+java_import 'org.apache.commons.math3.linear.CholeskyDecomposition'
+java_import 'MatrixGenerator'
+java_import 'ArrayGenerator'
+java_import 'MathHelper'
+java_import 'ArrayComparator'
+
+class NMatrix
+  include_package 'org.apache.commons.math3.analysis.function'
+  attr_accessor :shape, :dim, :dtype, :stype, :s
+
+  def initialize(*args)
+    if args[-1] == :copy
+      @shape = [2,2]
+      @s = [0,0,0,0]
+      @dim = shape.is_a?(Array) ? shape.length : 2
+    else
+      if (args.length <= 3)
+        @shape = args[0]
+        if args[1].is_a?(Array)
+          elements = args[1]
+          if args.length > 2
+            hash = args[2]
+            # puts hash
+            @dtype = hash[:dtype]
+            @stype = hash[:stype]
+          else
+            @dtype = :double
```

Comment:
I am not throwing any error here because a lot of tests will fail in that case. And I assure you that this feature will be properly implemented in next PR.